### ISSUE TITLE
Docker container support

### ISSUE THREAD

**Issue by toto99303** [drnick23/drplotter#44]
Hello, please add official docker container images for DrPlotter.
This is important as it allows multiple dockers to share the same GPUs, which is very convinient.
Gigahorse and noSSD already have their images and using them is very easy, so DrPlotter should not be missing this!
Thanks in advance!

**Comment by rkroelin** [comment]
I’d second this

**Comment by felixbrucker** [comment]
i made some docker images, cant test them right now, but give em a try: https://github.com/foxypool/drplotter/pkgs/container/drplotter
example compose: https://github.com/foxypool/drplotter/blob/main/docker-compose.yaml

**Comment by toto99303** [comment]
Thanks @felixbrucker , I will try them soon. I'm just not sure how to proceed with the initial token generation and also the ca folder? What would be the order of running the dockers if starting from scratch? Thanks again for you time!

**Comment by felixbrucker** [comment]
Off the top of my head you'll need to:
- generate a token: `docker compose run -it --rm solver --generate-token` and update the compose file with the generated token
- uncomment the `CHIA_CA` env var for the harvester and add the path to it into the volume section
- update the farmer ip/port env of the harvester to yours
- run the harvester once to init the ca: `docker compose run -it --rm harvester` and once that's done press ctrl+c to shut it down again and remove the env and volume mount for the ca dir from the compose file
- remove the services you don't need and start everything with `docker compose up -d`

ofc also the obvious stuff like adjusting the paths to your plot dirs etc

**Comment by toto99303** [comment]
I got the plotter running, but not sure how to deal with the cli output.
It's originally supposed to work in normal terminal, but here it's going to the live log and it looks like this:

![image](https://github.com/drnick23/drplotter/assets/20519170/36f49911-899b-46dc-83df-42517414d85b)

Progress, time and ETA not updating live...

**Comment by felixbrucker** [comment]
yeah i'm not sure about that, drplotter could detect that it doesn't have a tty, so it does its stdout differently, or its something else

**Comment by felixbrucker** [comment]
@drnick23 might have more insights on that
